app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsSheetDialogFragment.kt

```diff
@@ -151,10 +152,25 @@ class QuickSettingsSheetDialogFragment : AppCompatDialogFragment() {
     private fun arePermissionsGranted(requestCode: Int, grantResults: IntArray) =
         requestCode == REQUEST_CODE_QUICK_SETTINGS_PERMISSIONS && grantResults.all { it == PERMISSION_GRANTED }
 
-    private fun toggleTrackingProtection(context: Context, url: String) {
-        val host = url.tryGetHostFromUrl()
-        lifecycleScope.launch {
-            ExceptionDomains(context).toggle(host)
+    private fun toggleTrackingProtection(context: Context) {
+        val useCase = TrackingProtectionUseCases(
+            sessionManager = context.components.core.sessionManager,
+            engine = context.components.core.engine
+        )
+        val session = context.components.core.sessionManager.findSessionById(sessionId)
+        session?.let {
+            useCase.containsException(it) { contains ->
+                if (contains) {
+                    useCase.removeException(it)
```

Comment:
Should we  also track when the user remove an exception?

Reply:
We would need a new telemetry probe and data review

Reply:
I don't really know if we need  it just want to confirm :)